src/utils/mod.rs

```diff
@@ -0,0 +1,60 @@
+// Copyright (c) 2017 fd developers
+// Licensed under the Apache License, Version 2.0
+// <LICENSE-APACHE or http://www.apache.org/licenses/LICENSE-2.0>
+// or the MIT license <LICENSE-MIT or http://opensource.org/licenses/MIT>,
+// at your option. All files in the project carrying such
+// notice may not be copied, modified, or distributed except
+// according to those terms.
+
+use std::path::Path;
+use std::iter::Iterator;
+
+/// Determine if an os string ends with any of the given extensions (case insensitive).
+pub fn path_has_any_extension<'a, I>(path: &Path, exts: I) -> bool
+where
+    I: 'a + IntoIterator<Item = &'a String>,
+{
+    _path_has_any_extension(path, exts)
```

Comment:
Couldn't we just define two `path_has_any_extension` methods but where one has the `#[cfg(all(unix, not(target_os = "macos"))]` and the other has `#[cfg(any(target_os = "macos", windows))]` rather than wrapping a single platform specific method that does everything?

Reply:
This avoids duplicating the documentation comment for the two methods I guess :smile:

Reply:
That's good enough a reason for me